InciCsvCorrector.java

```diff
@@ -0,0 +1,159 @@
+
+    private static final int DATE_LENGTH = 10;  // dd/mm/yyyy
+    private static final int MINIMUM_VALID_LENGTH = 15; // of a line
+
+    public static void main(String[] args)
+    {
+        String[] paths = getFilesPaths();
+        File source = new File(paths[0]);
+        File corrected = new File(paths[1]);
+        boolean needsCorrection = false;
+        ArrayList<Integer> invLines = null;
+
+        try {
+            invLines = findInvalidLines(source);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        if(!(invLines.isEmpty())) {   // if there are invalid lines print their number
+            needsCorrection = true;
+            System.out.print("invalid lines : ");
+            while(!invLines.isEmpty()){
+                System.out.print(invLines.remove(0) + " ");
+            }
+            System.out.println();
+        }
+        if(needsCorrection){
+            try {
+                correctInterruptedLines(source, corrected);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            System.out.println("the file has been corrected");
+        }else{
+            System.out.println("the file does not need correction");
+        }
+    }
+
+    /**
+     * finds the invalid lines in the source file and prints them
+     * @param source source file
+     * @return an ArrayList of integers with the line numbers in the source file
+     * @throws IOException when the file does not exist
+     */
+    private static ArrayList<Integer> findInvalidLines(File source) throws IOException {
+
+        BufferedReader br = new BufferedReader(new FileReader(source));
+        String line;
+        int lineNum=0;
+        ArrayList<Integer> invalidLines = new ArrayList<>();
+
+        while (((line = br.readLine()) != null) ) {
+            lineNum++;
+            if(line.length()>MINIMUM_VALID_LENGTH) {
+                boolean currentLineHasValidStart = (isNumeric(line.substring(0, 5)) && (line.charAt(5) == ','));        //valid start of line (12345,)
+                boolean currentLineHasValidEnd = isDate(line.substring(line.length() - DATE_LENGTH));              //valid end of line (dd/mm/yyyy)
+                if (!(currentLineHasValidStart && currentLineHasValidEnd)) {               //adds to list the invalid lines
+                    invalidLines.add(lineNum);
+                }
+            }else{
+                invalidLines.add(lineNum);
+            }
+        }
+        return invalidLines;
+    }
+
+    /**
+     * eliminates line endings that break the ingredient entries writing the results in the corrected file
+     * @param source source file
+     * @param corrected result file
+     */
+    private static void correctInterruptedLines(File source,File corrected) throws IOException {
+
+        BufferedReader br = new BufferedReader(new FileReader(source));
+        BufferedWriter bw = new BufferedWriter(new FileWriter(corrected));
```

Comment:
Too short names for objects/variables can lead to a less readable code (mostly in more complex functions), you should be more explicit.